optuna/cli.py

```diff
@@ -51,6 +56,27 @@ def _check_storage_url(storage_url: Optional[str]) -> str:
     raise CLIUsageError("Storage URL is not specified.")
 
 
+def _get_storage(storage_url: Optional[str], storage_class: Optional[str]) -> BaseStorage:
+    storage_url = _check_storage_url(storage_url)
+    if storage_class:
+        if storage_class == JournalRedisStorage.__name__:
+            return JournalStorage(JournalRedisStorage(storage_url))
+        if storage_class == JournalFileStorage.__name__:
+            return JournalStorage(JournalFileStorage(storage_url))
+        if storage_class == RDBStorage.__name__:
+            return RDBStorage(storage_url)
+        raise CLIUsageError("Unsupported storage class")
+
+    if storage_url.startswith("redis"):
+        return JournalStorage(JournalRedisStorage(storage_url))
+    if os.path.isfile(storage_url):
+        return JournalStorage(JournalFileStorage(storage_url))
```

Comment:
More concretely, how about checking the existence of the files?
```suggestion
    if os.path.isfile(storage_url) or not os.path.exists(storage_url):
        return JournalStorage(JournalFileStorage(storage_url))
```

Reply:
In this suggested change, we are trying to make sure that the directory does not exist.
But won't it cause a bug if we specify a directory that doesn't exist?

Reply:
[to be discussed in a follow-up] Maybe we could check whether `os.path.dirname(storage_url)` is an existing directory?

Reply:
The usage of `os.path.dirname(storage_url)` might not provide a straightforward solution because in some cases, `os.path.dirname` treats a part of the DB_URL as a parent directory. Therefore, the issue may not be resolved with this approach alone.

```python
>>> import os
>>> os.path.dirname("sqlite:///foo/bar.db")
'sqlite:///foo'
```

Let us accept the current implementation and discuss it in a follow-up PR.